Document new option 'custom' for cloudflare integration's imageService. (#7813)
withastro/docs: src/content/docs/en/guides/integrations-guide/cloudflare.mdx

* **`cloudflare`:** Uses the [Cloudflare Image Resizing](https://developers.cloudflare.com/images/image-resizing/) service.
* **`passthrough`:** Uses the existing [`noop`](/en/guides/images/#configure-no-op-passthrough-service) service.
* **`compile`:** Uses Astro's default service (sharp), but only on pre-rendered routes at build time. During SSR for pages rendered on-demand, all `astro:assets` features are disabled.

> * **`custom`:** Always uses the image service configured in [Image Options](/en/reference/configuration-reference/#image-options). **This option will not check to see whether the configured image service works in Cloudflare's `workerd` runtime**

Reviewer: ```suggestion
* **`custom`:** Always uses the image service configured in [Image Options](/en/reference/configuration-reference/#image-options). **This option will not check to see whether the configured image service works in Cloudflare's `workerd` runtime.**
```

* **`cloudflare`:** Uses the [Cloudflare Image Resizing](https://developers.cloudflare.com/images/image-resizing/) service.
* **`passthrough`:** Uses the existing [`noop`](/en/guides/images/#configure-no-op-passthrough-service) service.
* **`compile`:** Uses Astro's default service (sharp), but only on pre-rendered routes at build time. During SSR for pages rendered on-demand, all `astro:assets` features are disabled.

> * **`custom`:** Always uses the image service configured in [Image Options](/en/reference/configuration-reference/#image-options)

Reviewer 2: ```suggestion
* **`custom`:** Always uses the image service configured in [Image Options](/en/reference/configuration-reference/#image-options). **This integration will not check to see whether the configured image service works in Cloudflare's `workerd` runtime**, so ....
```
I feel like this warning is better given *here*. (And, we try to avoid notes/asides etc. when we can just add the content right to the docs themselves where it makes sense)!

I will note, I don't know what it means that "guardlines are disabled". Is that something people will understand? Is that an important addition? Again, I'd rather see it *here* if it makes sense than in some other interjection far removed from the description itself.

Reviewer: So for all other options it checks for incompatible image services and replaces an incompatible service with the noop service. That doesn't happen for the custom option, that's what is meant with this warning.

Reviewer 2: Right, do the words "guardlines are disabled" convey any extra information beyond the existing "This integration will not check..."? Does it mean the same thing? Is it wording that Cloudflare users will be expecting to hear? That's my question! If it's just repeating the idea that there won't be any check, then it's like saying the same thing twice, right?

Reviewer: I think the updated version is good now.

> 

Reviewer: ```suggestion
```

})
```

:::note

> With  **`imageService: 'custom'`** guardlines are disabled for the configured image service. This integration will not check if the configured image service is functional in cloudflare environment.

Reviewer: ```suggestion
With  **`imageService: 'custom'`** guardlines are disabled for the configured image service. This integration will not check if the configured image service works in Cloudflare's `workerd` runtime.
```
